elementary/clients/slack/client.py

```diff
@@ -112,6 +118,14 @@ def send_report(self, channel: str, report_file_path: str):
             logger.error("Failed to send report to Slack.")
         return send_succeed
 
+    def get_user_id_from_email(self, email: str) -> str:
```

Comment:
Can you add a cache to this function? e.g. something like:
```
if email not in self.email_to_user_id_cache:
     self.email_to_user_id_cache[email] = self.client.users_lookupByEmail(email=email)["user"]["id"] 
return self.email_to_user_id_cache[email]
```

Mainly want to avoid having to wait several times for slack API calls on the same user for different alerts.